### PR TITLE
docs: step-debugging: add firewall tip at top, update Debug Configuration to use PHP Web Page, fixes #5248

### LINKED ISSUE #5248: Allow access to port 9003
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

_No response_

### Expected Behavior

Xdebug could connect.

### Actual Behavior

Xdebug: [Step Debug] Time-out connecting to debugging client, waited: 200 ms. Tried: host.docker.internal:9003 (fallback through xdebug.client_host/xdebug.client_port).

### Steps To Reproduce

Open port 9003 in the firewall.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by gitressa**
## The Issue

* #5248.

## Release/Deployment Notes

<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/5249"></a>

### CODE REVIEW

**Review by rfay — CHANGES_REQUESTED** [review]
Thanks for the contribution!

I can't find anything in https://stackoverflow.com/questions/64878376/xdebug-step-debug-could-not-connect-to-debugging-client (the link you provide) that has to do with firewall configuration. Also, not sure if we're going to be able to fully help people with this.

**Comment by gitressa** [comment]
Oops, sorry @rfay. I meant to add the link pointing to the section far down the page mentioning firewall. I updated the MR.

I do feel that users can waste inordinate amounts of time (I did) on many wild and unfruitful goose chases, but that this simple tip may save them from that.

**Comment by rfay** [comment]
What Linux distro were you on, and what firewall were you using there? I haven't ever had any trouble with Ubuntu. Often get reports of trouble on WSL2, where people may be running any number of firewalls.

**Comment by gitressa** [comment]
20.04 with ufw.

**Comment by rfay** [comment]
Thanks. I'm really surprised the ufw would interfere with docker in this configuration. I'll try it out.

File: docs/content/users/debugging-profiling/step-debugging.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ All IDEs basically work the same, listening on a port and reacting when they’r
 It will remain enabled until you start or restart the project.
 * Disable Xdebug for better performance when not debugging with `ddev xdebug off`.
 * `ddev xdebug status` will show Xdebug’s current status.
+* You may need to [open port 9003 in your firewall](https://stackoverflow.com/a/72591361) to allow Xdebug access. See also "Troubleshooting Xdebug" below.
```
Comment (mattstein) [inline docs/content/users/debugging-profiling/step-debugging.md:18]: You could save someone a few seconds linking directly to that section:

```suggestion
* You may need to [open port 9003 in your firewall](https://stackoverflow.com/a/72591361) to allow Xdebug access. (See [Troubleshooting Xdebug](#troubleshooting-xdebug).)
```

**Review by mattstein — CHANGES_REQUESTED** [review]

File: docs/content/users/debugging-profiling/step-debugging.md (hunk 1 above)
Comment (gitressa) [reply docs/content/users/debugging-profiling/step-debugging.md]: Thanks @mattstein, that's a great suggestion. Committed!

**Comment by rfay** [comment]
I confirm that on Debian/Ubuntu with just `sudo ufw enable` and no configuration ports are blocked for docker. 

So `sudo ufw disable` or `sudo ufw allow 9003` are required.

`ddev logs` will show something like this in that case:

```
[11-Aug-2023 00:05:41] WARNING: [pool www] child 2877 said into stderr: "NOTICE: PHP message: Xdebug: [Step Debug] Time-out connecting to debugging client, waited: 200 ms. Tried: 172.19.0.1:9003 (from HTTP_X_FORWARDED_FOR HTTP header), host.docker.internal:9003 (fallback through xdebug.client_host/xdebug.client_port)."
2023/08/11 00:05:41 [error] 2089#2089: *66 FastCGI sent in stderr: "PHP message: Xdebug: [Step Debug] Time-out connecting to debugging client, waited: 200 ms. Tried: 172.19.0.1:9003 (from HTTP_X_FORWARDED_FOR HTTP header), host.docker.internal:9003 (fallback through xdebug.client_host/xdebug.client_port)" while reading response header from upstream, client: 172.19.0.5, server: , request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "d9.ddev.site:8443"
[11-Aug-2023 00:05:41] WARNING: [pool www] child 2877 said into stderr: "NOTICE: PHP message: Xdebug: [Step Debug] Time-out connecting to debugging client, waited: 200 ms. Tried: 172.19.0.1:9003 (from HTTP_X_FORWARDED_FOR HTTP header), host.docker.internal:9003 (fallback through xdebug.client_host/xdebug.client_port).
```

**Comment by gitressa** [comment]
Thanks for confirming @rfay. Though I would definitely not recommend disabling the firewall.

**Comment by rfay** [comment]
@gitressa please take a look at this edited version and see if it resolves your concern. https://github.com/ddev/ddev/pull/5249/commits/581614164a0b8e689c6aca86d1319f471424155c

**Comment by gitressa** [comment]
It looks good, thanks @rfay and @mattstein.

**Pull request merged**